File: composer/src/DTOs/UIStageConfig.php
<?php

namespace ReactUbiquitous\DTOs;

use ReactUbiquitous\Contracts\SerializableInterface;
use ReactUbiquitous\Enums\UITheme;
use ReactUbiquitous\Enums\UIPageTransition;

final class UIStageConfig implements SerializableInterface
{
    /** @param UIPageConfig[] $pages */
    public function __construct(
        public readonly string $id,
        public readonly array $pages,
        public readonly ?string $title = null,
        public readonly ?string $description = null,
        public readonly ?string $defaultPageId = null,
        public readonly UITheme|string|null $theme = null,
        public readonly ?string $className = null,
        public readonly UIPageTransition|string|null $pageTransition = null,
    ) {}

    public function toArray(): array
    {
        return array_filter([
            'id' => $this->id,
            'title' => $this->title,
            'description' => $this->description,
            'defaultPageId' => $this->defaultPageId,
            'pages' => array_map(fn(UIPageConfig $p) => $p->toArray(), $this->pages),
            'theme' => $this->theme instanceof UITheme ? $this->theme->value : $this->theme,
            'className' => $this->className,
            'pageTransition' => $this->pageTransition instanceof UIPageTransition
                ? $this->pageTransition->value
                : $this->pageTransition,
        ], fn($v) => $v !== null);
    }
}

File: src/ReactUbiquitous.NuGet/Models/UIStageConfig.cs
using System.Text.Json.Serialization;

namespace ReactUbiquitous.NuGet.Models;

/// <summary>Represents the top-level configuration for a UI stage.</summary>
public class UIStageConfig
{
    /// <summary>Gets or sets the unique stage identifier.</summary>
    [JsonPropertyName("id")]
    public string Id { get; set; } = string.Empty;

    /// <summary>Gets or sets the stage title.</summary>
    [JsonPropertyName("title")]
    public string? Title { get; set; }

    /// <summary>Gets or sets the stage description.</summary>
    [JsonPropertyName("description")]
    public string? Description { get; set; }

    /// <summary>Gets or sets the identifier of the default page to display.</summary>
    [JsonPropertyName("defaultPageId")]
    public string? DefaultPageId { get; set; }

    /// <summary>Gets or sets the list of pages in the stage.</summary>
    [JsonPropertyName("pages")]
    public List<UIPageConfig>? Pages { get; set; }

    /// <summary>Gets or sets the visual theme for the stage.</summary>
    [JsonPropertyName("theme")]
    public string? Theme { get; set; }

    /// <summary>Gets or sets additional CSS class names for the stage.</summary>
    [JsonPropertyName("className")]
    public string? ClassName { get; set; }

    /// <summary>Gets or sets the page transition animation style.</summary>
    [JsonPropertyName("pageTransition")]
    public string? PageTransition { get; set; }
}
